pydis_site/apps/content/resources/guides/python-guides/app-commands.md

```diff
@@ -0,0 +1,417 @@
+
+## An example with a **Group** subclass!
+
+```python
+import discord
+from discord.ext import commands
+from discord import app_commands
+
+class MySlashGroup(app_commands.Group, name="uwu"):
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+        super().__init__()
+
+    @app_commands.command(name="ping", description="...")
+    async def _ping(self, interaction: discord.) -> None:
+        await interaction.response.send_message("pong!")
+
+    @app_commands.command(name="command", description="...")
+    async def _cmd(self, interaction: discord.Interaction) -> None:
+        await interaction.response.send_message("uwu")
+
+class MySlashBot(commands.Bot):
+    def __init__(self) -> None:
+        super().__init__(command_prefix="!", intents=discord.Intents.default())
+
+    async def setup_hook(self) -> None:
+        await self.add_cog(MySlashGroup(self))
+        await self.tree.copy_global_to(discord.Object(id=123456789098765432))
+        await self.tree.sync()
+
+bot = MySlashBot()
+
+bot.run("token")
+```
+
+__**EXPLANATION**__
+- The only difference here too is that the `MySlashGroup` class directly subclasses the **Group** class from discord.app_commands which automatically registers all the methods within the group class to be commands of that specific group. So now, the commands such as `ping` can be run using **/uwu ping** and `command` using **/uwu command**.
+
+# Some common methods and features used for Slash Commands.
+
+### A common function used for Slash Commands is the `describe` function. This is used to add descriptions to the arguments of a slash command. The command function can decorated with this function. It goes by the following syntax as shown below.
+
+```python
+from discord.ext import commands
+from discord import app_commands
+import discord
+
+bot = commands.Bot(command_prefix=".", intents=discord.Intents.default())
+#sync the commands
+
+@bot.tree.command(name="echo", description="...")
+@app_commands.describe(text="The text to send!", channel="The channel to send the message in!")
+async def _echo(interaction: discord.Interaction, text: str, channel: discord.TextChannel=None):
+    channel = channel or interaction.channel
+    await channel.send(text)
+```
+
+### Another common issue that most people come across is the time duration of sending a message with `send_message`. This issue can be tackled by deferring the interaction response using the `defer` method of the `InteractionResponse` class. An example for fixing this issue is shown below.
+
+```python
+import discord
+from discord.ext import commands
+import asyncio
+
+bot = commands.Bot(command_prefix="!", intents=discord.Intents.default())
+#sync the commands
+
+@bot.tree.command(name="time", description="...")
+async def _time(interaction: discord.Interaction, time_to_wait: int):
+    # -------------------------------------------------------------
+    await interaction.response.defer(ephemeral=True, thinking=True)
```

Comment:
```suggestion
    await interaction.response.defer(ephemeral=True)
```
We don't want to use `thinking` because if we don't use a followup the thinking state will never go away.
> thinking ([bool](https://docs.python.org/3/library/functions.html#bool)) – Indicates whether the deferred type should be [InteractionResponseType.deferred_channel_message](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=interactionresponse#discord.InteractionResponseType.deferred_channel_message) instead of the default [InteractionResponseType.deferred_message_update](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=interactionresponse#discord.InteractionResponseType.deferred_message_update) if both are valid. In UI terms, this is represented as if the bot is thinking of a response. It is your responsibility to eventually send a followup message via [Interaction.followup](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=interactionresponse#discord.Interaction.followup) to make this thinking state go away. Application commands (AKA Slash commands) cannot use [InteractionResponseType.deferred_message_update](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=interactionresponse#discord.InteractionResponseType.deferred_message_update).